### ISSUE TITLE
Differing behaviour in html5 output for an svg image with align specified

### ISSUE THREAD

**Issue by jtanx** [sphinx-doc/sphinx#7112]
**Describe the bug**

When an image is specified, similar to:

```rst
.. image:: the-image.svg
   :align: right
   :scale: 200px
```

This code: https://github.com/sphinx-doc/sphinx/blob/d4aeae475943aef9b935f7487baf90ccc1c58b42/sphinx/writers/html5.py#L552-L554

Sticks the svg (for unknown reasons) in a div and somehow also makes it clickable (so adds an internal reference, even though no `target` has been specified).

The clickable reference is added by this code:

https://github.com/sphinx-doc/sphinx/blob/d4aeae475943aef9b935f7487baf90ccc1c58b42/sphinx/builders/html.py#L808-L825

Apparently the clickable thing is a feature(?).

As it's specified in a div, this displayed in `block` style, which means content does not flow around it, but is displayed below the image. This is **different** to how alignment works on images.

e.g.:
![image](https://user-images.githubusercontent.com/5137410/74081639-6010e280-4aa5-11ea-848c-d3e7e837bac0.png)

![image](https://user-images.githubusercontent.com/5137410/74081643-7880fd00-4aa5-11ea-89c2-bee843405c14.png)

**To Reproduce**

Try rendering an svg to html5

```rst
.. image:: the-image.svg
   :align: right
   :scale: 200px
```

**Expected behavior**

The svg should behave no different to any other image - it should be styled inline and there should be no reference generated when no `target` is specified, e.g.:

![image](https://user-images.githubusercontent.com/5137410/74081649-8cc4fa00-4aa5-11ea-8e6b-534cf2a3289d.png)
![image](https://user-images.githubusercontent.com/5137410/74081651-92224480-4aa5-11ea-957b-7c6dd76b85d8.png)

**Your project**
N/A

**Screenshots**

**Environment info**
- OS: Windows
- Python version: ] 3.7.2
- Sphinx version: 2.3.1
- Sphinx extensions:  sphinx.ext.mathjax
- Extra tools: Firefox

**Additional context**
N/A

**Comment by tk0miya** [comment]
Thank you for reporting. I posted #7114 to fix this.

Note: Sphinx does not apply any hyperlinks to SVG images because it can have its own hyperlinks. This is difference between other images formats.

**Comment by jtanx** [comment]
Awesome, thanks! Would you know the reason why SVGs need to be wrapped in divs, instead of just being treated like other images?

**Comment by tk0miya** [comment]
Good point. Originally, HTML builder uses `<object>` tag to embed SVG tag. That is why we use `<div>` tag for alignment. Now, we use `<img>` tag for embedding. So `<div>` tag is no longer needed... It's time for refactoring!

**Comment by tk0miya** [comment]
I updated my PR not to use `<div>` tag for alignment.
It will be released as 2.4.0 (maybe released in this weekend).
Thanks,
